test/source-sink-errors/mzcompose.py

```diff
@@ -103,27 +187,48 @@ def populate(self, c: Composition) -> None:
             )
         )
 
-    def assert_error(self, c: Composition, error: str) -> None:
+    def assert_error(self, c: Composition, error: str, hint: Optional[str]) -> None:
```

Comment:
do we need this, if we are overriding `assert_error` above anyways? Removing it would resolve Philip's concern below!

Reply:
Do you mean removing the option to match `hint` for the previously existing redpanda tests? (And yup, it would resolve https://github.com/MaterializeInc/materialize/pull/18009#discussion_r1136611844)

Reply:
Reverted the Redpanda specific changes!